interceptor.go

```diff
@@ -235,17 +235,45 @@ func (i *interceptor) WrapStreamingClient(next connect.StreamingClientFunc) conn
 			protocol: protocolToSemConv(conn.Peer()),
 			attrs:    attributesFromRequest(req),
 		}
+		carrier := propagation.HeaderCarrier(conn.RequestHeader())
```

Comment:
This probably shouldn't be done here; request headers _shouldn't_ technically be allowed to be passed in from clients